modules/afsocket/afsocket-source.c

```diff
@@ -758,13 +795,13 @@ afsocket_sd_restore_kept_alive_connections(AFSocketSourceDriver *self)
       GList *p = NULL;
       self->connections = cfg_persist_config_fetch(cfg, afsocket_sd_format_connections_name(self));
 
-      self->num_connections = 0;
+      _connections_count_set(self, 0);
```

Comment:
When `keep-alive()` is disabled, the counter does not work properly.
This is because the lifetime of the stats counter is different than the drivers'.

Reproduction: reload syslog-ng when it has active connections with `keep-alive(no)`. The counter should reset to 0 in this case.  An explicit `_connections_count_set(self, 0);` call is needed during the initialization of the driver.